packages/api/src/task/scheduler.ts

```diff
@@ -167,11 +174,13 @@ export class TaskScheduler {
 
   private async failTask(task: Task, error: string, output?: Task["output"]) {
     const status = {
+      ...task.status,
```

Comment:
@victorges I added these on all the status updates to avoid loss of the attempts tracking, although I am not sure of all the implications. Is this correct from your POV or is there something that could cause other fields in the status to be overwritten or preserved when they shouldn't?

Reply:
WDYT of doing something similar to this on the `updateTask` function?
https://github.com/livepeer/livepeer-com/blob/ae8791b43052ad4339f73389e73c5cd241c01e9b/packages/api/src/task/scheduler.ts#L327-L330

i.e. Clients don't need to worry about merging the previous status with the new status nor even of setting an `updatedAt`, as the update func will handle that.